src/renderer/ipcRegistration.ts

```diff
@@ -123,3 +123,28 @@ ipc.on('initiate-export-project', async () => {
 
   store.dispatch(startExport(currentProject.id, filePath));
 });
+
+/**
+ * Used by backend to initiate return to home from front end
+ */
+window.electron.on('initiate-return-to-home', async () => {
+  const { currentProject } = store.getState();
+  const saveChanges = 0;
+
+  if (currentProject === null) return;
+
+  const userSelection = await window.electron.returnToHome(currentProject);
+
+  // if user wants to save unsaved changes
+  if (userSelection === saveChanges) {
+    const filePath = await window.electron.saveProject(currentProject);
+
+    const projectMetadata = await window.electron.retrieveProjectMetadata({
+      ...currentProject,
+      projectFilePath: filePath,
+    });
+
+    store.dispatch(projectSaved(currentProject, projectMetadata, filePath));
+  }
+  store.dispatch(pageChanged(ApplicationPage.HOME));
```

Comment:
I think you also need to dispatch a projectClosed (or whatever it's called) action to set the currentProject to null in the store, otherwise the currentProject will still be set which has consequences for save buttons, file representation etc

Reply:
Updated!